docs/source/gateway/index.rst

```diff
@@ -1,6 +1,1049 @@
+A provider represents the source of the machine learning models, such as OpenAI, Anthropic, and so on.
+Each provider has its specific characteristics and configurations that are encapsulated within the model part of a route in the MLflow AI Gateway.
+
+Supported Provider Models
+^^^^^^^^^^^^^^^^^^^^^^^^^
+The table below presents a non-exhaustive list of models and a corresponding route type within the MLflow AI Gateway.
+With the rapid development of LLMs, there is no guarantee that this list will be up to date at all times. However, the associations listed
+below can be used as a helpful guide when configuring a given route for any newly released model types as they become available with a given provider.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Route Type
+     - Provider
+     - Model Examples
+     - Supported
+   * - llm/v1/completions
+     - OpenAI
+     - gpt-3.5-turbo, gpt-4
+     - ✓
+   * - llm/v1/completions
+     - Anthropic
+     - claude-1, claude-1.3-100k
+     - ✓
+   * - llm/v1/completions
+     - Cohere
+     - command, command-light-nightly
+     - ✓
+   * - llm/v1/completions
+     - Azure OpenAI
+     - text-davinci-003, gpt-35-turbo
+     - ✓
+   * - llm/v1/chat
+     - OpenAI
+     - gpt-3.5-turbo, gpt-4
+     - ✓
+   * - llm/v1/chat
+     - Anthropic
+     -
+     - ✗
+   * - llm/v1/chat
+     - Cohere
+     -
+     - ✗
+   * - llm/v1/chat
+     - Azure OpenAI
+     - gpt-35-turbo, gpt-4
+     - ✓
+   * - llm/v1/embeddings
+     - OpenAI
+     - text-embedding-ada-002
+     - ✓
+   * - llm/v1/embeddings
+     - Anthropic
+     -
+     - ✗
+   * - llm/v1/embeddings
+     - Cohere
+     - embed-english-v2.0, embed-multilingual-v2.0
+     - ✓
+   * - llm/v1/embeddings
+     - Azure OpenAI
+     - text-embedding-ada-002
+     - ✓
+
+Within each model block in the configuration file, the provider field is used to specify the name
+of the provider for that model. This is a string value that needs to correspond to a provider the MLflow AI Gateway supports.
+
+Here's an example of a provider configuration within a route:
+
+.. code-block:: yaml
+
+    routes:
+        - name: chat
+          type: llm/v1/chat
+          model:
+            provider: openai
+            name: gpt-4
+            config:
+              openai_api_base: https://api.openai.com/v1
+              openai_api_key: $OPENAI_API_KEY
+
+In the above configuration, ``openai`` is the `provider` for the model.
+
+As of now, the MLflow AI Gateway supports the following providers:
+
+* **openai**: This is used for models offered by `OpenAI <https://platform.openai.com/>`_.
+* **anthropic**: This is used for models offered by `Anthropic <https://docs.anthropic.com/claude/docs>`_.
+* **cohere**: This is used for models offered by `Cohere <https://docs.cohere.com/docs>`_.
+* **azure** / **azuread**: This is used for model services offered by `Azure <https://learn.microsoft.com/en-gb/azure/cognitive-services/openai/>`_.
```

Comment:
Nit: These aren't providers, they're just `openai_api_type` specs for `openai`. Can we instead say that `**openai**` (line 420) supports OpenAI, Azure OpenAI, and Azure OpenAI with AAD?